### ISSUE TITLE
risky-file-permissions: not triggered in nested block (E208)

### ISSUE THREAD

**Issue by kwevers** [ansible/ansible-lint#1008]
##### Summary

If eg a copy task is missing the mode attribute and it is in a nested block statement E208 is not triggered.
The same task in a single block statement or at the root level does trigger E208.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible 2.9.12
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/id853249/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/id853249/.local/lib/python3.6/site-packages/ansible
  executable location = /home/id853249/.local/bin/ansible
  python version = 3.6.8 (default, Dec  5 2019, 15:45:45) [GCC 8.3.1 20191121 (Red Hat 8.3.1-5)]
ansible-lint --version
ansible-lint 4.3.1
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

cat /etc/redhat-release && uname -r
Red Hat Enterprise Linux release 8.2 (Ootpa)
4.18.0-193.14.3.el8_2.x86_64

##### STEPS TO REPRODUCE

```console (paste below)
---
- hosts: localhost
  tasks:
    - name: Simple copy
      copy:
        src: src
        dest: dest

- hosts: localhost
  tasks:
    - block:
        - name: Copy in block
          copy:
            src: src
            dest: dest

- hosts: localhost
  tasks:
    - block:
        - block:
            - name: Copy in nested block
              copy:
                src: src
                dest: dest

```

##### Desired Behaviour

I expect all 3 tasks to trigger E208

##### Actual Behaviour

Only the first two tasks trigger E208

```paste below
$ ansible-lint test.yml 
[208] File permissions not mentioned
test.yml:4
Task/Handler: Simple copy

[208] File permissions not mentioned
test.yml:12
Task/Handler: Copy in block

You can skip specific rules by adding them to the skip_list section of your configuration file:                         

┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│# .ansible-lint                                                                                                       │
│skip_list:                                                                                                            │
│  - '208'  # File permissions not mentioned'                                                                          │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
@kwevers A patch would be more than welcomed here.

**Comment by netzvieh** [comment]
Seems to me that nothing is triggered once you are two layers deep in blocks. So this has to be handled on the task extraction level.

**Comment by mehes-kth** [comment]
> Seems to me that nothing is triggered once you are two layers deep in blocks. So this has to be handled on the task extraction level.

That is _exactly_ correct.  I was about to file a similar bug report.

[Here](https://github.com/ansible-community/ansible-lint/blob/master/src/ansiblelint/utils.py#L667-L673)'s the culprit:
```python
    # Add sub-elements of block/rescue/always to tasks list
    tasks.extend(extract_from_list(tasks, ['block', 'rescue', 'always']))
    # Remove block/rescue/always elements from tasks list
    block_rescue_always = ('block', 'rescue', 'always')
    tasks[:] = [
        task for task in tasks if all(k not in task for k in block_rescue_always)
    ]
```
First, all top-level `block` nodes are replaced by their children.
Then, all `block` nodes (including those newly added from any child) are deleted.

**Comment by ssbarnea** [comment]
@mehes-kth We no longer have priority labels, they make no sense in a community project. If you find it critical please make a PR to fix it and I will be glad to help you merge and release a it.

Due to historical reasons lots of rules still have parsing inside them, which in fact we should slowly eradicate. Still, fixing this properly is much harder than you may thing, especially because we would need to normalize the task tree and copy inhered properties from upper levels (blocks, playbook). Once this is done we should be able to look at a single task and make a decision. For the moment the only workaround I used was to be explicit on each task.

**Comment by mehes-kth** [comment]
The little I've seen of this code base, I do understand that revamping anything takes major effort.
That's also the prime reason for finding it difficult to meaningfully contribute.  Trying to convince
our team to stop using `ansible-lint` would be effort better spent, I feel.  So, no PR from me...

I understand that the update to the title is intended as a clarification (using text instead of a code),
but it's making the problem seem much milder than it actually is.  As I pinpointed in the relevant
code snippet above, **nothing** is triggered in **any** block that's nested inside another.
I can file a separate bug report on that with the right title and description, if you like.

**Comment by ssbarnea** [comment]
Does not reproduce with recent versions.
